cpp/src/quantiles/quantiles.cu

```diff
@@ -16,196 +16,253 @@
       }
   }
     
   struct trampoline_exact_functor{
     template <typename T,
               typename std::enable_if_t<!std::is_arithmetic<T>::value, int> = 0>
     gdf_error operator()(gdf_column* col_in,
-                         gdf_quantile_method prec,
-                         double q,
-                         void* t_erased_res,
-                         gdf_context* ctxt)
+                         gdf_quantile_method interpolation,
+                         double              quant,
+                         void*               t_erased_res,
+                         gdf_context*        ctxt,
+                         cudaStream_t        stream = NULL)
     {
       return GDF_UNSUPPORTED_DTYPE;
     }
 
     template <typename T,
               typename std::enable_if_t<std::is_arithmetic<T>::value, int> = 0>
-    gdf_error operator()(gdf_column*  col_in,              //input column;
-                         gdf_quantile_method prec,         //precision: type of quantile method calculation
-                         double              q,            //requested quantile in [0,1]
-                         void*               t_erased_res, //result; for <exact> should probably be double*; it's void* because
-                                                           //(1) for uniformity of interface with <approx>;
-                                                           //(2) for possible types bigger than double, in the future;
-                         gdf_context*        ctxt)         //context info
+    gdf_error operator()(gdf_column*  col_in,
+                         gdf_quantile_method interpolation,
+                         double              quant,
+                         void*               t_erased_res,
+                         gdf_context*        ctxt,
+                         cudaStream_t        stream = NULL)
     {
-      return trampoline_exact<T>(col_in, prec, q, t_erased_res, ctxt);
+      // just in case double won't be enough to hold result
+      // it can be changed in future
+      return trampoline_exact<T, double>
+                 (col_in, interpolation, quant, t_erased_res, ctxt, stream);
     }
   };
 
   struct trampoline_approx_functor{
     template <typename T>
-    void operator()(gdf_column*  col_in,       //input column;
-                    double       q,            //requested quantile in [0,1]
-                    void*        t_erased_res, //type-erased result of same type as column;
-                    gdf_context* ctxt)         //context info)
+    gdf_error operator()(gdf_column*  col_in, 
+                    double       quant,
+                    void*        t_erased_res,
+                    gdf_context* ctxt,
+                    cudaStream_t stream = NULL)
     {
-      trampoline_approx<T>(col_in, q, t_erased_res, ctxt);
+      return trampoline_exact<T, T>(col_in, GDF_QUANT_LOWER, quant, t_erased_res, ctxt, stream);

```

Comment:
Hmmm, I thought you were going to eliminate all references to "approx" since approx is just exact but with GDF_QUANT_LOWER interpolation, and Pandas doesn't have "approx", it just has different types of interpolation. Is this not the plan?

Reply:
This is my plan: https://github.com/rapidsai/cudf/pull/1416#issuecomment-483986087

Reply:
OK, sounds good.